packages/lab/src/overlay/Overlay.tsx

```diff
@@ -75,51 +61,42 @@ export const Overlay = forwardRef<HTMLDivElement, OverlayProps>(
       ],
     });
 
-    const { getReferenceProps, getFloatingProps } = useInteractions([
-      useRole(context, { role: "dialog" }),
-      useClick(context),
-      useDismiss(context),
-    ]);
+  const { getReferenceProps, getFloatingProps } = useInteractions([
+    useRole(context, { role: "dialog" }),
+    useClick(context),
+    useDismiss(context),
+  ]);
 
-    const floatingStyles = useMemo(() => {
-      return {
-        top: y ?? 0,
-        left: x ?? 0,
-        position: strategy,
-        width: elements.floating?.offsetWidth,
-        height: elements.floating?.offsetHeight,
-      };
-    }, [elements.floating, strategy, x, y]);
-
-    const arrowProps = {
-      ref: arrowRef,
-      context,
+  const floatingStyles = useMemo(() => {
+    return {
+      top: y ?? 0,
+      left: x ?? 0,
+      position: strategy,
+      width: elements.floating?.offsetWidth,
+      height: elements.floating?.offsetHeight,
     };
+  }, [elements.floating, strategy, x, y]);
 
-    const handleCloseButtonClick = () => {
-      handleOpenChange(false);
-    };
+  const arrowProps = {
+    ref: arrowRef,
+    context,
+  };
 
-    return (
-      <OverlayContext.Provider
-        value={{
-          id: id ?? "",
-          openState,
-          floatingStyles,
-          placement,
-          context,
-          arrowProps,
-          floating,
-          reference,
-          handleCloseButtonClick,
-          getFloatingProps,
-          getReferenceProps,
-        }}
-      >
-        <div ref={ref} {...rest}>
```

Comment:
I don't think ref and {...rest} here are needed and therefore removed forward ref.
This component is just providing context, a ref and styles can still be passed down directly to the OverlayPanel or OverlayTrigger component - can't think of a scenario where it would need to be passed down to the Overlay Component directly